app/controllers/campaign_pages_controller.rb

```diff
@@ -23,6 +23,14 @@ def create
     i = 0
     widgets.each do |widget_type_name, widget_data|
       widget_type_id = widget_data.delete('widget_type')
+
+
+      if defined? widget_data['checkboxes']['{cb_number}']
```

Comment:
I see what your trying to do, but I was trying to say that ```defined?``` doesn't do what you think it does. It's not testing for the presence of that nested hash.